src/Covenant/Data.hs

```diff
@@ -260,9 +265,25 @@ data DatatypeInfo
       Show
     )
 
+mkDatatypeInfo :: DataDeclaration AbstractTy -> DatatypeInfo AbstractTy
+mkDatatypeInfo decl = DatatypeInfo decl baseFStuff (mkBBF decl)
+  where
+    baseFStuff :: Maybe (DataDeclaration AbstractTy, ValT AbstractTy)
+    baseFStuff =
+      let baseFDecl = runReader (mkBaseFunctor decl) 0
+          baseBBF = mkBBF =<< baseFDecl
+       in (,) <$> baseFDecl <*> baseBBF
+
+renameDatatypeInfo :: DatatypeInfo AbstractTy -> Either RenameError (DatatypeInfo Renamed)
+renameDatatypeInfo (DatatypeInfo ogDecl baseFStuff bb) = runRenameM $ do
+  ogDecl' <- renameDataDecl ogDecl
+  baseFStuff' <- traverse (bitraverse renameDataDecl renameValT) baseFStuff
+  bb' <- traverse renameValT bb
+  pure $ DatatypeInfo ogDecl' baseFStuff' bb'
```

Comment:
I would say this belongs together with `renameValT` and `renameCompT`, which are in `Rename`. Is there a reason why this lives here?